src/prometheus/registry.zig

```diff
@@ -71,6 +72,75 @@ pub fn Registry(comptime options: RegistryOptions) type {
             self.arena_state.deinit();
         }
 
+        /// Initialize a struct full of metrics.
+        /// Every field must be a supported metric type.
+        pub fn initStruct(self: *Self, comptime Struct: type) GetMetricError!Struct {
+            var metrics_struct: Struct = undefined;
+            inline for (@typeInfo(Struct).Struct.fields) |field| {
+                try self.initMetric(Struct, &@field(metrics_struct, field.name), field.name);
+            }
+            return metrics_struct;
+        }
+
+        /// Initialize any fields within the struct that are supported metric types.
+        /// Leaves other fields untouched.
+        ///
+        /// Returns the number of fields that were *not* initialized.
+        pub fn initFields(
+            self: *Self,
+            /// Mutable pointer to a struct containing metrics.
+            metrics_struct: anytype,
+        ) GetMetricError!usize {
+            const Struct = @typeInfo(@TypeOf(metrics_struct)).Pointer.child;
+            const fields = @typeInfo(Struct).Struct.fields;
+            var num_fields_skipped: usize = fields.len;
+            inline for (@typeInfo(Struct).Struct.fields) |field| {
```

Comment:
Debug logging mean you need to pass in a logger whenever you call this, which is doable but adds some dependency burden.

I made this function return a usize with the total number of skipped fields, so you can assert it in the calling context to ensure you're not missing anything. That ends up looking something like this:

```zig
        var self: Self = undefined;
        const registry = globalRegistry();
        std.debug.assert(try registry.initFields(&self) == 1);
        self._logging_fields = .{ .logger = logger };
```

I feel like the assertion approach is pretty good at detecting whether you're missing some fields. It's not perfect but if you add new fields and forget to init them in the struct, you'll get an error pretty quickly when you run the code. I could also make it a comptime check if desired, the tradeoff would be mean you *always* have to run the check, and don't have the option to ignore it. I went with the assertion approach because I liked the flexibility and it seemed good enough.